pkg/rtc/mediatrack.go

```diff
@@ -349,23 +358,41 @@ func (t *MediaTrack) closeChan() {
 }
 
 func (t *MediaTrack) updateStats() {
+
 	for {
 		select {
 		case <-t.done:
 			return
 		case <-time.After(connectionQualityUpdateInterval):
-			t.statsLock.Lock()
+			t.connectionStats.Lock.Lock()
+
+			//we are accessing stats after receiver has buffer assigned
+			stats := t.buffer.GetStats()
+
+			delta := t.connectionStats.UpdateStats(stats.TotalByte)
 			if t.Kind() == livekit.TrackType_AUDIO {
-				t.connectionStats.CalculateAudioScore()
+				t.connectionStats.Score = connectionquality.AudioConnectionScore(delta, t.connectionStats.Jitter)
 			} else {
-				t.calculateVideoScore()
+				t.connectionStats.Score = t.calculateVideoScore()
+			}
+			stat := &livekit.AnalyticsStat{
+				Jitter:          float64(t.connectionStats.Jitter),
+				TotalPackets:    uint64(t.connectionStats.TotalPackets),
+				PacketLost:      uint64(t.connectionStats.PacketsLost),
+				Delay:           uint64(t.connectionStats.Delay),
+				TotalBytes:      t.connectionStats.TotalBytes,
+				NackCount:       t.connectionStats.NackCount,
+				PliCount:        t.connectionStats.PliCount,
+				FirCount:        t.connectionStats.FirCount,
+				ConnectionScore: float32(t.connectionStats.Score),
 			}
-			t.statsLock.Unlock()
+			t.params.Telemetry.TrackStats(livekit.StreamType_UPSTREAM, t.PublisherID(), t.ID(), stat)
```

Comment:
Are the counters supposed to be all cumulative over time or should connection stats be reset after posting or maybe some fields need to be reset?

Reply:
stats like totalBytes,totalPackets,totalPacketsLost would be better if they are cumilative, as that would allow us flexibility in the future to use these stats (server side retries ..etc). Time sensitive stats like jitter, delay ..etc need not be